src/common/utils.h

```diff
@@ -713,6 +713,23 @@ inline void EmplaceBackZeros(const NDArrayStorageType stype, const TShape &shape
   }
 }
 
+
+/*!
+ * \brief parallelize copy by OpenMP.
+ */
+template<typename DType>
+inline void ParallelCopy(DType* dst, const DType* src, index_t size) {
+  static index_t copy_block_size = dmlc::GetEnv("MXNET_CPU_PARALLEL_COPY_SIZE", 200000);
+  if (size >= copy_block_size) {
+    #pragma omp parallel for num_threads(engine::OpenMP::Get()->GetRecommendedOMPThreadCount())
+    for (index_t i = 0; i < size; ++i) {
+      dst[i] = src[i];
+    }
+  } else {
+    std::memcpy(dst, src, sizeof(DType) * size);
```

Comment:
There is a CheckContinuous function call before the memcpy in original mshadow::Copy function (https://github.com/dmlc/mshadow/blob/696803bd7723ade8230af878460d96c68a550fbc/mshadow/tensor_cpu-inl.h#L132)

Do we not need this any more?

Reply:
Hi, I don't think CheckContinuous function is necessary. here are two reasons:
1. class TBlob seems to be continuous always. because of [CheckContiguous function in TBlob class](https://github.com/apache/incubator-mxnet/blob/14ae5d4ef09ce4334603da9ad32af1f697ad6961/include/mxnet/tensor_blob.h#L170-L172);
2. the inputs (from, to) of TBlob type will convert to Tensor(by `to->FlatTo1D()`) in the primitive implement of ndarray::Copy<cpu,cpu>. accroding to the [code of FlatTo1D](https://github.com/apache/incubator-mxnet/blob/14ae5d4ef09ce4334603da9ad32af1f697ad6961/include/mxnet/tensor_blob.h#L211-L216), [get_with_shape](https://github.com/apache/incubator-mxnet/blob/14ae5d4ef09ce4334603da9ad32af1f697ad6961/include/mxnet/tensor_blob.h#L283-L294),[constructor of tensor](https://github.com/dmlc/mshadow/blob/696803bd7723ade8230af878460d96c68a550fbc/mshadow/tensor.h#L445-L448) and [CheckContiguous function in Tensor](https://github.com/dmlc/mshadow/blob/696803bd7723ade8230af878460d96c68a550fbc/mshadow/tensor.h#L473-L475), when convert TBlob input to Tensor, Tensor's stride_ is assigned by `shape[dim - 1]`, And the `CheckContiguous function` in Tensor always return true. there is no need to check this.